kusto/test/etoe/ingestion_status_e2e_test.go

```diff
@@ -0,0 +1,395 @@
+
+	client, err := kusto.New(testConfig.Endpoint, testConfig.Authorizer)
+	if err != nil {
+		panic(err)
+	}
+
+	ingestor, err := ingest.New(client, testConfig.Database, tableName)
+	if err != nil {
+		panic(err)
+	}
+
+	count := 5
+	var ch [5]chan error
+	var errors [5]error
+
+	for i := 0; i < count; i++ {
+		f, err := os.Open(csvFile)
+		if err != nil {
+			panic(err)
+		}
+
+		reader, writer := io.Pipe()
+		go func() {
+			defer writer.Close()
+			io.Copy(writer, f)
+		}()
+
+		res, err := ingestor.FromReader(ctx, reader, ingest.ReportResultToTable(), ingest.FileFormat(ingest.CSV))
+		ch[i] = res.Wait(ctx)
+	}
+
+	for i := 0; i < count; i++ {
+		errors[i] = <-ch[i]
+	}
+
+	for i, err := range errors {
+		if err != nil {
+			t.Errorf("Exepcted status Succeeded however result on channel %d is:\n%s", i+1, err)
+		}
+	}
+}
+
+func TestIgestionFromFileWithoutStatusReporting(t *testing.T) {
+	err := initOnce()
+	if err != nil {
+		t.Skipf("Skipping tests: %s", err)
+		return
+	}
+
+	ctx, cancel := context.WithCancel(context.Background())
+	ctx, cancel = context.WithTimeout(ctx, 1*time.Minute)
+	defer cancel()
+
+	client, err := kusto.New(testConfig.Endpoint, testConfig.Authorizer)
+	if err != nil {
+		panic(err)
+	}
+
+	ingestor, err := ingest.New(client, testConfig.Database, tableName)
+	if err != nil {
+		panic(err)
+	}
+
+	res, err := ingestor.FromFile(ctx, csvFile)
+	if err != nil {
+		panic(err)
+	}
+
+	err = <-res.Wait(ctx)
+	if err != nil {
+		t.Errorf("Exepcted status Queued however result is:\n%s", err)
+	}
+}
+
+func TestIgestionFromReaderWithoutStatusReporting(t *testing.T) {
+	err := initOnce()
+	if err != nil {
+		t.Skipf("Skipping tests: %s", err)
+		return
+	}
+
+	ctx, cancel := context.WithCancel(context.Background())
+	ctx, cancel = context.WithTimeout(ctx, 1*time.Minute)
+	defer cancel()
+
+	client, err := kusto.New(testConfig.Endpoint, testConfig.Authorizer)
+	if err != nil {
+		panic(err)
+	}
+
+	ingestor, err := ingest.New(client, testConfig.Database, tableName)
+	if err != nil {
+		panic(err)
+	}
+
+	f, err := os.Open(csvFile)
+	if err != nil {
+		panic(err)
+	}
+
+	reader, writer := io.Pipe()
+	go func() {
+		defer writer.Close()
+		io.Copy(writer, f)
+	}()
+
+	res, err := ingestor.FromReader(ctx, reader, ingest.FileFormat(ingest.CSV))
+	if err != nil {
+		panic(err)
+	}
+
+	err = <-res.Wait(ctx)
+	if err != nil {
+		t.Errorf("Exepcted status Queued however result is:\n%s", err)
+	}
+}
+
+func TestIgestionFromFileWithClientFailure(t *testing.T) {
+	err := initOnce()
+	if err != nil {
+		t.Skipf("Skipping tests: %s", err)
+		return
+	}
+
+	ctx, cancel := context.WithCancel(context.Background())
+	ctx, cancel = context.WithTimeout(ctx, 1*time.Minute)
+	defer cancel()
+
+	client, err := kusto.New(testConfig.Endpoint, testConfig.Authorizer)
+	if err != nil {
+		panic(err)
+	}
+
+	ingestor, err := ingest.New(client, testConfig.Database, tableName)
+	if err != nil {
+		panic(err)
+	}
+
+	// Once without status reporting

```

Comment:
Make this table driven, your doing double tests of the same thing with an option.

Reply:
I disagree with this test philosophy.
Tests should check one case and report it.
Table driven fails on first test and does not uncover further issues.

Reply:
For instance it's very hard for me right now to debug a specific case that does not ingest well in the old e2e code.
I have to let the loop run it's course and check which case is the relevant one.

Reply:
The etoe tests are not a really great version of Go testing.  It was a rough conversion of some test code that was written in a different coding style.  It roughly fits in here.  So there are ugly edges here.

Table driven tests are the standard Go method of testing.  Where it makes sense (testing more than one option), we should default to how everyone in Go does things. 

Table driven tests only fail on the first test if you choose for it to do so.  A table driven test can tests all cases and report only on ones that fail.  Not to mention there is a whole way of doing subtests (though I don't use this facility much, probably because it didn't exist when I started).  

However, at this point, I think the etoe needs the next stage of love to go from the, "translate from .Net testing to rough Go testing", and I can just do this in a later CL.